omod/src/main/java/org/bahmni/module/hip/web/controller/BundledMedicationRequestController.java

```diff
@@ -26,19 +33,32 @@ public BundledMedicationRequestController(BundleMedicationRequestService bundled
 
     @RequestMapping(method = RequestMethod.GET, value = "/medication", produces = MediaType.APPLICATION_JSON_VALUE)
     public @ResponseBody
-    ResponseEntity<String> getBundledMedicationRequestFor(@RequestParam(required = false) String patientId,
-                                                          @RequestParam(required = false) String visitType) {
-
-        if (patientId == null || patientId.equals("''"))
-            throw new RequestParameterMissingException("patientId");
+    ResponseEntity<?> getBundledMedicationRequestFor(@NotNull @NotEmpty @RequestParam(required = false) String patientId,
+                                                     @NotNull @NotEmpty @RequestParam(required = false) String visitType,
+                                                     @RequestHeader(value = "Authorization", required = false) String basicAuthorization) {
+        if (isValidUser(basicAuthorization)) {
+            if (patientId == null || patientId.equals(""))
+                throw new RequestParameterMissingException("patientId");
 
-        if (visitType == null || visitType.equals("''"))
-            throw new RequestParameterMissingException("visitType");
+            if (visitType == null || visitType.equals(""))
+                throw new RequestParameterMissingException("visitType");
 
-        Bundle bundle = bundledMedicationRequestService.bundleMedicationRequestsFor(patientId, visitType);
-        return ResponseEntity.ok()
-                .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
-                .body(serializeBundle(bundle));
+            Bundle bundle = bundledMedicationRequestService.bundleMedicationRequestsFor(patientId, visitType);
+            return ResponseEntity.ok()
+                    .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
+                    .body(serializeBundle(bundle));
+        }
+        return ResponseEntity.status(HttpStatus.UNAUTHORIZED).body(ClientError.unauthorizedUser());
     }
 
-}
\ No newline at end of file
+    private boolean isValidUser(String basicAuthorization) {
+        if (basicAuthorization != null && basicAuthorization.startsWith("Basic")) {
+            String base64Credentials = basicAuthorization.substring("Basic".length()).trim();
+            byte[] credDecoded = Base64.getDecoder().decode(base64Credentials);
+            String credentials = new String(credDecoded, StandardCharsets.UTF_8);
+            final String[] values = credentials.split(":", 2);
+            return values[0].equals("superman") && values[1].equals("Admin123");
```

Comment:
Username/password is not universally constant and cannot be included in the code like here